public/app/plugins/datasource/prometheus/querybuilder/components/metric-encyclopedia/MetricEncyclopediaModal.test.tsx

```diff
@@ -146,17 +137,17 @@ describe('MetricEncyclopediaModal', () => {
     expect(metricInsideRange).toBeInTheDocument();
   });
 
-  it('paginates millions of metrics and does not run out of memory', async () => {
-    const millionsOfMetrics: string[] = [...Array(1000000).keys()].map((i) => '' + i);
-    setup(defaultQuery, millionsOfMetrics);
+  it('paginates lots of metrics and does not run out of memory', async () => {
```

Comment:
While this asserts that 100000 elements can be re-paginated without crashing the test execution env, it doesn't really assert that lots of metrics won't crash the browser. 

I'd really like to see some e2e tests providing a bit more coverage on this, maybe we spend our next hack day adding some cypress tests around this (and maybe other UI components that we know are prone to break for high cardinality users)?

Reply:
Agreed, next hack day I will commit to doing e2e tests!